scipy/spatial/_geometric_slerp.py

```diff
@@ -0,0 +1,224 @@
+from __future__ import division, print_function, absolute_import
+
+__all__ = ['geometric_slerp']
+
+import numpy as np
+from scipy.spatial.distance import euclidean
+
+def _geometric_slerp(start, end, t):
+    # calculate the angle between `start` and `end`
+    s = np.prod(np.linalg.svd([start, end])[1])
+    c = np.dot(start, end)
+    omega = np.arctan2(s, c)
+
+    # create an orthogonal basis using QR decomposition
+    basis = np.vstack([start, end])
+    Q = np.linalg.qr(basis.T)[0].T
+    signs = np.sign(np.einsum('ij,ij->i', basis, Q))
+    start, end = Q * np.array([signs]).T
+
+    # interpolate
+    s = np.sin(t * omega)
+    c = np.cos(t * omega)
+    return start * c[:, np.newaxis] + end * s[:, np.newaxis]
+
```

Comment:
I have given my code some more thought. The SVD is not necessary since we do a QR decomposition. Here is the simplified code:
```
def _geometric_slerp(start, end, t):
    # create an orthogonal basis using QR decomposition
    basis = np.vstack([start, end])
    Q, R = np.linalg.qr(basis.T)
    signs = np.sign(np.diag(R))
    Q = Q.T * signs.T[:, np.newaxis]
    R = R.T * signs.T[:, np.newaxis]
    start, end = Q

    # calculate the angle between `start` and `end`
    c = np.dot(start, end)
    s = np.linalg.det(R)
    omega = np.arctan2(s, c)

    # interpolate
    s = np.sin(t * omega)
    c = np.cos(t * omega)
    return start * c[:, np.newaxis] + end * s[:, np.newaxis]
```
The relationship between the SVD and the QR is described here:
https://en.wikipedia.org/wiki/QR_decomposition#Connection_to_a_determinant_or_a_product_of_eigenvalues

Reply:
This doesn't pass the test suite at the moment---in particular, `TestGeometricSlerp.test_include_ends` fails, suggesting that `start` or `end` isn't being included anymore.

Looking a little more closely, it seems that `end` is excluded here now.

Reply:
In the meanwhile, I've added `asv` benchmarks, so that if you correct that code & stand by the claim that it is more efficient we can test the performance.

At the moment, `asv run -b 'GeometricSlerp*'`:

```
[100.00%] ··· ....GeometricSlerpBench.time_geometric_slerp_3d                 ok
[100.00%] ··· ============ =========
               num_points
              ------------ ---------
                   10       321±3μs
                  1000      381±2μs
                 10000      819±4μs
              ============ =========
```

Reply:
Thanks. There was indeed a bug in the code. Here is the fixed code:
```
def _geometric_slerp(start, end, t):
    # create an orthogonal basis using QR decomposition
    basis = np.vstack([start, end])
    Q, R = np.linalg.qr(basis.T)
    signs = np.sign(np.diag(R))
    Q = Q.T * signs.T[:, np.newaxis]
    R = R.T * signs.T[:, np.newaxis]

    # calculate the angle between `start` and `end`
    c = np.dot(start, end)
    s = np.linalg.det(R)
    omega = np.arctan2(s, c)

    # interpolate
    start, end = Q
    s = np.sin(t * omega)
    c = np.cos(t * omega)
    return start * c[:, np.newaxis] + end * s[:, np.newaxis]
```
Benchmarks on my laptop:
```
============ ============ ============
 num_points    with svd     qr only
------------ ------------ ------------
     10       39.0±0.3ms   31.2±0.4ms
    1000       46.7±1ms    39.4±0.2ms
   10000       114±1ms     95.9±0.3ms
============ ============ ============
```
It is faster but not by much, so pick the version you prefer.

I wrapped the benchmark in a for loop to get millisecond rather than nanosecond times:
```
    def time_geometric_slerp_3d(self, num_points):
        # time geometric_slerp() for 3D interpolation
        for i in range(100):
            geometric_slerp(start=self.start,
                            end=self.end,
                            t=self.t)
```

Reply:
Ok, I pulled in your most recent implementation, though conventional benchmarks show that both approaches are effectively the same within the default performance threshold.